pytest_operator/plugin.py

```diff
@@ -98,6 +105,15 @@ def pytest_configure(config):
     )
 
 
+def pytest_runtest_setup(item):
```

Comment:
I think the name `pytest_runtest_setup` inherited from [here](https://github.com/pytest-dev/pytest/blob/0ad452bcbadffa2bb890c543c8b409d2df61c67d/doc/en/example/markers.rst#custom-marker-and-command-line-option-to-control-test-runs) can be tailored to our use.  I believe it is called `runtest_setup` because it is a setting to decide if it runs tests (see the helps/comments in the example).  Ours would be slightly different (something to do with skipping if deployed?).

If we had two different conditions (say `skip_if_deployed` and `run_if_XXX`), would we implement both in this function or two separate functions?  I don't understand exactly how this works so am not sure, but that would influence how it should be named (and maybe a docstring explaining what it does would help too)

Reply:
> If we had two different conditions (say skip_if_deployed and run_if_XXX), would we implement both in this function or two separate functions? I don't understand exactly how this works so am not sure, but that would influence how it should be named (and maybe a docstring explaining what it does would help too)


Both should be in this function. There is this [runtest protocol](https://docs.pytest.org/en/7.0.x/reference/reference.html?highlight=pytest_runtest_setup#pytest.hookspec.pytest_runtest_protocol), where the `pytest_runtest_setup` is in from setup phase. I understand that it is similar to `setUp` function from `unittest.TestCase`.

I will add docstring to this function with explanation and reference to docs + examples.